src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -477,7 +485,7 @@ private static function scrapeEqualityAssertions(
     }
 
     /**
-     * @param PhpParser\Node\Expr\BinaryOp\NotIdentical|PhpParser\Node\Expr\BinaryOp $conditional
+     * @param PhpParser\Node\Expr\BinaryOp\NotIdentical|PhpParser\Node\Expr\BinaryOp\NotEqual $conditional
```

Comment:
I recall there's quite a bit of dead code in this class. It's not caught by Psalm because of param types being too wide.

It could be interesting to implement param counterpart to 'make return types as tight as possible' option, at least for private/protected methods.

Reply:
I'd kill for that feature :p

I tried everything, I used Psalter to fix MissingParamType while telling Psalm to ignore phpdoc, I tried to tweak the code to make Psalm fix MissingParamType even when the Param is already there but it's a mess every time

Reply:
I narrowed some phpdoc, mainly for $conditional. Let's see if there's new issues

Reply:
I think it was `hasPositiveNumberCheck()` and `hasZeroCheck()` methods. They are only ever called from `getGreaterAssertions()`, but have code handling Identical, Smaller, Equal etc.

Reply:
Right, I missed quite a few. I'm on it

Reply:
Done. There was some dead code that I removed. I wonder if I should leave a note to this PR for future readers that may need to reimplement the code